e2e/config/azure.go

```diff
@@ -600,3 +608,74 @@ func (a *AzureClient) createVMSS(ctx context.Context, resourceGroupName string,
 	return &vmssResp.VirtualMachineScaleSet, nil
 
 }
+
+// GetLatestVMExtensionImageVersion lists VM extension images for a given extension name and returns the latest version.
+// This is equivalent to: az vm extension image list -n Compute.AKS.Linux.AKSNode --latest
+func (a *AzureClient) GetLatestVMExtensionImageVersion(ctx context.Context, location, extType, extPublisher string) (string, error) {
+	// List extension versions
+	resp, err := a.VMExtensionImages.ListVersions(ctx, location, extPublisher, extType, &armcompute.VirtualMachineExtensionImagesClientListVersionsOptions{})
+	if err != nil {
+		return "", fmt.Errorf("listing extension versions: %w", err)
+	}
+
+	if len(resp.VirtualMachineExtensionImageArray) == 0 {
+		return "", fmt.Errorf("no extension versions found")
+	}
+
+	version := make([]Version, len(resp.VirtualMachineExtensionImageArray))
+	for i, ext := range resp.VirtualMachineExtensionImageArray {
+		version[i] = parseVersion(ext)
+	}
+
+	sort.Slice(version, func(i, j int) bool {
+		return version[i].Less(version[j])
+	})
+
+	return *version[len(version)-1].Original.Name, nil
+}
+
+// Version represents a parsed version of a VM extension image.
+type Version struct {
+	Original *armcompute.VirtualMachineExtensionImage
+	Major    int
+	Minor    int
+	Patch    int
+}
+
+// parseVersion parses the version from a VM extension image name, which can be in the format 1.151, 1.0.1, etc.
+// You can find all the versions of a specific VM extension by running:
+// az vm extension image list -n Compute.AKS.Linux.AKSNode
+func parseVersion(v *armcompute.VirtualMachineExtensionImage) Version {
+	// Split by dots
+	parts := strings.Split(*v.Name, ".")
+
+	version := Version{Original: v}
+
+	if len(parts) >= 1 {
+		if major, err := strconv.Atoi(parts[0]); err == nil {
+			version.Major = major
+		}
+	}
+	if len(parts) >= 2 {
+		if minor, err := strconv.Atoi(parts[1]); err == nil {
+			version.Minor = minor
+		}
+	}
+	if len(parts) >= 3 {
+		if patch, err := strconv.Atoi(parts[2]); err == nil {
+			version.Patch = patch
+		}
+	}
+
+	return version
+}
+
+func (v Version) Less(other Version) bool {
```

Comment:
can we use functionality within `semver` instead?

Reply:
Do you mean use the `semver` package? If you see the list of the versions, it is not consistent. Here is a sample output:

```bash
➜  az vm extension image list -n Compute.AKS.Linux.AKSNode | jq -r '.[] | .version' | sort -V
1.0.0.4
1.0.1
1.0.3
1.0.4.1
1.8
1.9
1.10
1.12
1.13
1.16
1.17
1.22
1.23
1.25
1.26
1.27
1.29
1.30
1.33
1.39
1.43
1.45
1.46
1.48
1.51
1.56
1.57
1.58
1.60
1.61
1.62
1.67
1.69
1.70
1.73
1.75
1.83
1.94
1.103
1.105
1.108
1.110
1.113
1.126
1.130
1.132
1.135
1.141
1.146
1.149
1.151
1.154
1.170
1.175
1.183
1.191
1.194
1.195
1.196
1.204
1.208
1.210
1.218
1.224
1.231
1.234
1.239
1.244
1.245
1.253
1.256
1.260
1.268
1.271
```